Pull mozilla vpn attribution from fxa events

relud opened the pull request:
Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title)
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated

relud commented on sql/moz-fx-data-shared-prod/mozilla_vpn_derived/login_flows_v1/query.sql:
```diff
@@ -13,11 +13,38 @@ WITH base AS (
+        STRUCT(
+          entrypoint_experiment,
+          entrypoint_variation,
+          CAST(NULL AS STRING) AS referrer,
```
blocking this on figuring out referrer with @birdsarah

birdsarah replied:
I don't think fxa_email_first - view is the right event any more. Users will land on a subscription platform page. Not the accounts login page.

relud replied:
> I don't think fxa_email_first - view is the right event any more. Users will land on a subscription platform page. Not the accounts login page.

Okay, I'll rework the use of that in the funnel as a separate issue.

relud commented:
@acmiyaguchi I'll need to make some changes when we figure out referrer, but requesting an early review on the rest of it. 

Explanation:

The login_flows_v1 table is derived from the fxa events that will continue to have attribution after Aug 25th. The users_v1 table is where we are currently getting attribution. I modified mozilla_vpn_derived.users_v1 to pull in attribution from login_flows_v1, and changed the type of the attribution field from a string map to a struct because this new data path requires attribution fields to be pre-defined.

I also modified the fxa_content_auth_events view to expose additional attribution fields currently included in users_v1, and `NULL`ed an unused field in mozilla_vpn_external.users_v1 that I noticed contained unhashed `fxa_uid`s.

acmiyaguchi commented on sql/moz-fx-data-shared-prod/mozilla_vpn_derived/all_subscriptions_v1/query.sql:
```diff
@@ -94,11 +94,11 @@ attribution AS (
-    mozfun.map.get_key(attribution, "utm_source") AS utm_source,
-    mozfun.map.get_key(attribution, "utm_content") AS utm_content,
-    mozfun.map.get_key(attribution, "utm_campaign") AS utm_campaign,
+    attribution.referrer,
```
Curious why this no longer requires the get_key construct. Was this introduced in another PR?

relud replied:
the underlying data type changed to a struct

acmiyaguchi reviewed (commented):
Can you quickly summarize the changes made in the PR for future reference? It looks like:

* removal of `active_subscriptions`, `new_subscriptions`, and retention_by_subscription` tables
* addition of `fxa_attribution` table
* removal of `attribution_category` and `coarse_attribution_category` fields
* addition of `entrypoint_experiment`, `entrypoint_variation`, and `utm_term` fields

Let me know if I missed anything. Also what's the plan for backfill?

The changes themselves look sane (other than the single test failure). r+ with passing CI.
